cmake/CompilerChecks.cmake

```diff
@@ -23,6 +23,10 @@ if(NOT DEFINED CMAKE_INTERPROCEDURAL_OPTIMIZATION)
     endif()
 endif()
 
+try_compile(HAVE_PTHREAD_TIMEDJOIN_NP ${PROJECT_BINARY_DIR}
+            ${PROJECT_SOURCE_DIR}/cmake/check_pthread_timedjoin_np.cpp
+            LINK_LIBRARIES Threads::Threads)
+
```

Comment:
Why not
```suggestion
include(CheckCXXSymbolExists)
check_cxx_symbol_exists(pthread_timedjoin_np pthread.h HAVE_PTHREAD_TIMEDJOIN_NP)
```

Reply:
That was my first attempt, but getting it to correctly use the pthread library was proving both difficult and ugly...  `try_compile` makes it easy to add the appropriate library flags.